contracts/helpers/TransferHelper.sol

```diff
@@ -324,36 +215,79 @@ contract TransferHelper is
         ) {
             // Check if the value returned from the external call matches
             // the conduit `execute` selector.
-            if (
-                conduitMagicValue != ConduitInterface(conduit).execute.selector
-            ) {
+            if (conduitMagicValue != ConduitInterface.execute.selector) {
                 // If the external call fails, revert with the conduit key
                 // and conduit address.
                 revert InvalidConduit(conduitKey, conduit);
             }
-        } catch (bytes memory data) {
-            // Catch reverts from the external call to the conduit and
-            // "bubble up" the conduit's revert reason.
-            revert ConduitErrorRevertBytes(data, conduitKey, conduit);
         } catch Error(string memory reason) {
             // Catch reverts with a provided reason string and
             // revert with the reason, conduit key and conduit address.
             revert ConduitErrorRevertString(reason, conduitKey, conduit);
+        } catch (bytes memory data) {
+            // Conduits will throw a custom error when attempting to transfer
+            // native token item types or an ERC721 item amount other than 1.
+            // Bubble up these custom errors when encountered. Note that the
+            // conduit itself will bubble up revert reasons from transfers as
+            // well, meaning that these errors are not necessarily indicative of
+            // an issue with the item type or amount in cases where the same
+            // custom error signature is encountered during a conduit transfer.
+
+            // Set initial value of first four bytes of revert data to the mask.
+            bytes4 customErrorSelector = bytes4(0xffffffff);
+
+            // Utilize assembly to read first four bytes (if present) directly.
+            assembly {
+                // Combine original mask with first four bytes of revert data.
+                customErrorSelector := and(
+                    mload(add(data, 0x20)), // Data begins after length offset.
+                    customErrorSelector
+                )
+            }
+
+            // Pass through the custom error in question if the revert data is
+            // the correct length and matches an expected custom error selector.
+            if (
+                data.length == 4 &&
```

Comment:
Nit: `data.length >= 4` is more appropriate here. Having extra data in ABI standard is still valid. But fine keeping it to stay consistent with Seaport's use of stricter ABI standard.

Reply:
Yeah also don’t want to risk false positives on errors with the same signature but with arguments